frontend/src/components/LoRAManager.tsx

```diff
@@ -0,0 +1,222 @@
+        <p className="text-xs text-muted-foreground">
+          No LoRA adapters configured. Add LoRA files to models/lora directory.
+        </p>
+      )}
+
+      <div className="space-y-2">
+        {loras.map(lora => (
+          <div
+            key={lora.id}
+            className="rounded-lg border bg-card p-3 space-y-2"
+          >
+            <div className="flex items-center justify-between gap-2">
+              <div className="flex-1">
+                <FilePicker
+                  value={lora.path}
+                  onChange={path => handleLoraChange(lora.id, { path })}
+                  files={availableLoRAs}
+                  disabled={disabled || isStreaming}
+                  placeholder="Select LoRA file"
+                  emptyMessage="No LoRA files found"
+                />
+              </div>
+              <Button
+                size="sm"
+                variant="ghost"
+                onClick={() => handleRemoveLora(lora.id)}
+                disabled={disabled || isStreaming}
+                className="h-6 w-6 p-0"
+                title={
+                  isStreaming
+                    ? "Cannot remove LoRAs while streaming"
+                    : "Remove LoRA"
+                }
+              >
+                <X className="h-3 w-3" />
+              </Button>
+            </div>
+
+            <div className="flex items-center gap-2">
+              <span className="text-xs text-muted-foreground w-12">Scale:</span>
+              <TooltipProvider>
+                <Tooltip>
+                  <TooltipTrigger asChild>
+                    <div className="flex-1">
+                      <SliderWithInput
+                        value={localScales[lora.id] ?? lora.scale}
+                        onValueChange={value => {
+                          handleLocalScaleChange(lora.id, value);
+                        }}
+                        min={-10}
+                        max={10}
+                        step={0.1}
+                        disabled={
+                          disabled ||
+                          (isStreaming &&
+                            loraMergeStrategy === "permanent_merge")
+                        }
+                        className="flex-1"
+                      />
+                    </div>
+                  </TooltipTrigger>
+                  <TooltipContent>
+                    <p className="text-xs">
+                      {isStreaming && loraMergeStrategy === "permanent_merge"
+                        ? "Runtime adjustment is disabled with Permanent Merge strategy. LoRA scales are fixed at load time."
+                        : "Adjust LoRA strength. Click Apply to update during streaming. 0.0 = no effect, 1.0 = full strength"}
+                    </p>
+                  </TooltipContent>
+                </Tooltip>
+              </TooltipProvider>
+            </div>
+          </div>
+        ))}
+      </div>
+
+      {hasUnsavedScales && (
```

Comment:
<img width="362" height="177" alt="CleanShot 2025-11-19 at 11 43 57" src="https://github.com/user-attachments/assets/454b6501-b80b-4940-a650-6b1c17ec6317" />

I found it a bit cumbersome to have to press another button after moving the slider. Can we just send the new scale value when the slider thumb stops like what we do for other sliders?

Reply:
> A third reason I just remembered: clicking the + and - buttons sends a bunch of updates

IIUC that's how the Denoising Step List and Noise Scale updates work too. Do you see that behavior as a negative?